### DISCUSSION TITLE
Can I get a callback from the client for drive redirection?

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8762] [Q&A]
Hello, I'm making a project using a sample client.

Can I get data on drive redirection from the client?

I want to get moving file data from drive redirection.

Is this possible? 

Also, I would appreciate it if you could tell me the location of the document related to drive redirection.

I looked for the X11 client, but I couldn't find the part about redirecting the drive separately on the client, so I wrote it. 

Thank you for reading the question.

**Comment by akallabeth** [comment]
`-v`? what do you want exactly?
as for the drive redirection specification, that is [MS-RDPEFS](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/34d9de58-b2b5-40b6-b970-f82d4603bdb5)

**Reply by kimcoldwater** [reply]
When implementing clipboard redirection on clients, I understand that local backend must be implemented.

And I was able to get the data of that response, and I was able to get the clipboard data into Java using JNI.  (like the Android version)

Like this, I want to import data from drive redirection to Java using JNI.


`android_cliprdr.c` `xf_cliprdr.c`
Like these things.

**Reply by akallabeth** [reply]
drive redirection redirects some local folder to the remote end. nothing to import.
are you maybe talking about file cliplboard (e.g. copy & paste files)?

**Reply by kimcoldwater** [reply]
No. I'm making RDP functionality available on the web, and I'm checking if drive redirection is available on the web.

I thought the data was moving like a clipboard, but I think I misunderstood.

I'm trying to create a file transfer function through drive redirection like Apache Guacamole.

Thanks to your answer, I thought of another way. 

Thank you for your help.

**Reply by akallabeth** [reply]
you might have to reimplement `rdpdr` then, because currently it assumes you can access files via `fopen` (or more precise `CreateFile` `API`)

**Reply by kimcoldwater** [reply]
I think I can access the file through fopen.

I will take note of your advice and test it.

Thank you.
